src/Microsoft.ML.AutoML/Experiment/MetricsAgents/RankingMetricsAgent.cs

```diff
@@ -10,11 +10,16 @@ internal class RankingMetricsAgent : IMetricsAgent<RankingMetrics>
     {
         private readonly MLContext _mlContext;
         private readonly RankingMetric _optimizingMetric;
+        private readonly int _dcgTruncationLevel;
 
-        public RankingMetricsAgent(MLContext mlContext, RankingMetric optimizingMetric)
+        public RankingMetricsAgent(MLContext mlContext, RankingMetric metric, int optimizationMetricTruncationLevel)
         {
             _mlContext = mlContext;
-            _optimizingMetric = optimizingMetric;
+            _optimizingMetric = metric;
+
+            // We want to make sure we always have at least 10 results. Getting extra results adds no measurable performance
+            // impact, so err on the side of more.
+            _dcgTruncationLevel = System.Math.Max(10, 2 * optimizationMetricTruncationLevel);
```

Comment:
Can you verify AutoML is optimizing the `@k` given by the user? I see from the unit test it's now reporting at least `@10`.

Reply:
Per offline discussion it is optimizing the @k from the user.

Reply:
Thanks for the fun pair programming session to get this fixed. Enjoyable as always.